endpoint/crypto/include_pub/privmx/endpoint/crypto/CryptoApi.hpp

```diff
@@ -118,6 +119,58 @@ class CryptoApi {
      */
     std::string convertPEMKeytoWIFKey(const std::string& pemKey);
 
+    /**
+     * Generate ECC key and BIP-39 mnemonic from a password using BIP-39
+     * 
+     * @param strength size of BIP-39 entropy if must be multiple 32
+     * @param password the password used to generate the Key
+     * @return BIP-39_t struct containing ECC Key and associated with it BIP-39 mnemonic and entropy
+     */
+    BIP39_t generate(std::size_t strength, const std::string& password = std::string());
```

Comment:
I think the name of the method "generate" is not descriptive
```suggestion
    BIP39_t generateBip39(std::size_t strength, const std::string& password = std::string());
```